htlcswitch/htlcnotifier.go

```diff
@@ -471,3 +489,14 @@ func getEventType(pkt *htlcPacket) HtlcEventType {
 		return HtlcEventTypeForward
 	}
 }
+
+// newPayHash returns the payment hash for the package provided.
+func newPayHash(pkt *htlcPacket) lntypes.Hash {
+	// In case the circuit is nil a zero payment hash is returned to satisfy
```

Comment:
Had to add the zero payment hash here, because some unit tests failed because the circuit was sometimes nil.